test_communication/test/test_subscription_valid_data.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <iostream>
+#include <string>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <test_communication/msg/u_int32.hpp>
```

Comment:
It's unrelated to this pr, but I'd say the transformation should not put an underscore between capital letters. `u_int32` is pretty silly looking, it should just be `uint32`. On the other hand, consider `RGBDImage` (obviously a manufactured example), which by my suggestion would be `rgbdimage`, should probably be `rgbd_image`. But even more wrong would be `r_g_b_d_image`. Obviously making the "correct" translation is not straight forward.